src/components/FilterSizes.js

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import size from '../img/size.png';
+import './style/FilterSizes.css';
+
+
+
+
+class FilterSizes extends React.Component {
+    //when you hover over a div, all others become visible and text pops up
+    hover = event => {
+        const popups = document.getElementsByClassName("popup");
+        for(let i = 0; i < popups.length; i++) {
+            popups[i].style.opacity = 0.9;
+        }
+        event.target.style.opacity = 0.1;
+        switch(event.target.id) {
+            case "one":
+                document.getElementById("input").style.zIndex = "3";
+                document.getElementById("input").style.opacity = 1;
+                break;
+            case "two":
+                document.getElementById("convolve").style.zIndex = "3";
+                document.getElementById("convolve").style.opacity = 1;
+                break;
+            case "three":
+                document.getElementById("filters").style.zIndex = "3";
+                document.getElementById("filters").style.opacity = 1;
+                break;
+            case "four":
+                document.getElementById("output-separate").style.zIndex = "3";
+                document.getElementById("output-separate").style.opacity = 1;
+                break;
+            case "five":
+                document.getElementById("output-together").style.zIndex = "3";
+                document.getElementById("output-together").style.opacity = 1;
+                break;
+            default:
+                break;
+        }
+    }
+
+    //when you leave the div, all divs become transparent
+    leave = () => {
+        const popups = document.getElementsByClassName("popup");
+        const text = document.getElementsByClassName("popup-text-container");
+        for(let i = 0; i < popups.length; i++) {
+            popups[i].style.opacity = 0;
+        }
+        for(let i = 0; i < text.length; i++) {
+            
+            text[i].style.opacity = 0;
+            text[i].style.zIndex="auto";
+        }
+
+
+    }
```

Comment:
If you follow the state-based hover approach, you'll want to change this as well. Bright side is it should be a performance gain since you won't need to double-loop!